utils/watermark_disk.go

```diff
@@ -60,8 +65,8 @@ func ApplyWatermarks(logicalName string) (func() error, error) {
 			}
 			// Check if the watermark is still in the disk
 			if slices.Equal(currentValue, watermark.data) {
-				ErrorExistingWatermark := errors.New("Error existing watermark in the file")
-				return fmt.Errorf("%s | %w", logicalName, ErrorExistingWatermark)
+				ErrorExistingWatermark := errors.New("Error existing watermark in the position: ")
+				return fmt.Errorf("%s | %w %d", logicalName, ErrorExistingWatermark, watermark.position)
```

Comment:
maybe `"%s@%d | %w"` instead? Otherwise the location is going to be after the error string when it should be closer to the device name I think.